crates/red_knot_test/src/parser.rs

```diff
@@ -136,12 +137,13 @@ pub(crate) struct EmbeddedFile<'s> {
 /// Matches an arbitrary amount of whitespace (including newlines), followed by a sequence of `#`
 /// characters, followed by a title heading, followed by a newline.
 static HEADER_RE: LazyLock<Regex> =
-    LazyLock::new(|| Regex::new(r"^(\s*\n)*(?<level>#+)\s+(?<title>.+)\s*\n").unwrap());
+    LazyLock::new(|| Regex::new(r"^(?<level>#+)\s+(?<title>.+)\s*\n").unwrap());
```

Comment:
Minor: the doc-comment should be updated ("Matches an arbitrary amount of whitespace"…)